bundles/org.eclipse.equinox.common/src/org/eclipse/core/runtime/IAdapterFactory.java

```diff
@@ -69,8 +72,23 @@ public interface IAdapterFactory {
 	 * types are supported, in advance of dispatching any actual
 	 * <code>getAdapter</code> requests.
 	 * </p>
+	 * <p>
+	 * The default implementation collects the required classes from the
+	 * {@link AdapterTypes} annotation, if that is not used implementors must
+	 * override this method.
+	 * </p>
 	 *
 	 * @return the collection of adapter types
 	 */
-	public Class<?>[] getAdapterList();
+	default Class<?>[] getAdapterList() {
+		Class<? extends IAdapterFactory> clz = getClass();
```

Comment:
In the best case it is actually never called, but as its in the interface we can't make sure, this can become quite annoying if one uses extension points (or service properties) and both list get out of sync the adapter is never called what can be quite "surprising" if you stare at them and see its there but never used ....